File: 1287-DistanceBetweenBusStops/1287-DistanceBetweenBusStops.java
// Last updated: 3/26/2026, 1:50:12 AM
class Solution {
    public int distanceBetweenBusStops(int[] distance, int start, int destination) {
        int sum1 = 0;
        int sum2 = 0;
        if(start > destination){
            int temp = start;
            start = destination;
            destination = temp;
        }
        
        for(int i=0;i<distance.length;i++){
            if(i>=start && i<destination)
                sum2+=distance[i];
            sum1+=distance[i];
        }
        return Math.min(sum1-sum2,sum2);
    }
}

File: 1287-DistanceBetweenBusStops/1287-DistanceBetweenBusStops.cs
// Last updated: 3/26/2026, 1:50:09 AM
public class Solution {
    public int DistanceBetweenBusStops(int[] distance, int start, int destination) {
        int atobDistance = 0;
        int btoaDistance = 0;
        
        int tempStart = Math.Min(start,destination);
        int tempDestination = Math.Max(start,destination);
        
        start = tempStart;
        destination = tempDestination;
        
        for(int i=start;i<destination;i++)
        {
            atobDistance+=distance[i];
        }
        
        for(int i=destination; ((i)%distance.Length)!=start;i++)
        {
            btoaDistance+=distance[i%distance.Length];
        }
        
        return Math.Min(atobDistance, btoaDistance);
    }
}